index.html

```diff
@@ -49,19 +49,19 @@ <h1 class="large l-align-center l-mb-mega">&lt;gaia-button&gt;</h1>
   <hr/>
 
   <section>
-    <gaia-button><i data-icon="back-light" aria-hidden="true"></i><span>Back</span></gaia-button>
-    <gaia-button><span>Forward</span><i data-icon="forward-light" aria-hidden="true"></i></gaia-button>
+    <gaia-button><span data-l10n-id="back" data-icon="back-light">Back</span></gaia-button>
+    <gaia-button><span data-l10n-id="forward" data-icon-end="forward-light">Forward</span></gaia-button>
   </section>
 
   <hr/>
 
   <section flex>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="bug"></i></gaia-button>
-    <gaia-button circular><i data-icon="video"></i></gaia-button>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="reload"></i></gaia-button>
-    <gaia-button circular><i data-icon="airplane"></i></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="bug" data-icon="bug"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="video" data-icon="video"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="reload" data-icon="reload"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="airplane" data-icon="airplane"/></gaia-button>
```

Comment:
According to zibi's feedback https://bugzilla.mozilla.org/show_bug.cgi?id=1162138#c16
l20n can't recognize which content element to attach with in gaia-element, so we need attach `data-l10n-id` on proper content element. So does data-icon.